x-pack/legacy/plugins/reporting/server/browsers/chromium/driver_factory/index.ts

```diff
@@ -46,23 +47,30 @@ export class HeadlessChromiumDriverFactory {
     this.binaryPath = binaryPath;
     this.browserConfig = browserConfig;
     this.queueTimeout = queueTimeout;
-    this.logger = logger;
     this.networkPolicy = networkPolicy;
+
+    this.userDataDir = fs.mkdtempSync(path.join(os.tmpdir(), 'chromium-'));
+    this.getChromiumArgs = (viewport: BrowserConfig['viewport']) =>
+      args({
+        userDataDir: this.userDataDir,
+        viewport,
+        disableSandbox: this.browserConfig.disableSandbox,
+        proxy: this.browserConfig.proxy,
+      });
```

Comment:
I didn't entirely get the performance improvements I was hoping for with this PR. I think the next thing to try would be to call `puppeteerLaunch` in the constructor of `DriverFactory`, and cache it as `this.browser`. Then the core of `createPage` would be: `this.browser.newPage()`

I think that would be a more major change, because it means different triggers for things like cleaning up the user data directory, etc.